### ISSUE TITLE
BCFtools filter - Issue with DP and basic question

### ISSUE THREAD

**Issue by KunathBJ** [samtools/bcftools#1209]
Hello,

I'm just starting using this suite of tool so sorry in advance if the questions might seem dumb.
I'm working with bacteria and I ran mpileup using multiple bam files (different time point of the same sample) and then wanted to filter the output based on the DP parameter.
So I used this command:
`bcftools mpileup -d 500 -Ou -f ref.fa file1.bam file2.bam file3.bam | \
bcftools call -Ou -mv --ploidy 1 | \
bcftools filter -s LowQual -e 'QUAL<20 && DP<10' > filtered.vcf`

Even though I used those particular filter, I could still see in the vcf file, some variant that passed the filter even thought they had a DP < 10
> contig_1        100     .       C       T       29.1805 PASS    DP=2;VDB=0.58;SGB=0.231553;MQ0F=0;AC=1;AN=1;DP4=0,0,2,0;MQ=60   GT:PL   1:57,0  .:0,0   .:0,0

How is that possible?
Also, my logical here for filtering is that I would like at least some coverage supporting the variant to consider it true.
But I read some example of filter where they exclude DP>100. Why is that? I don't understand the logical.

Finally, following the coverage reasoning, I would like to know how many reads from each bam file are supporting the variant I've found. But I cannot find any option (I must be missing it) in the bcftools suite that print for example the DP and the number of reads from each bam file.

Thanks a lot for your help.
I hope I was clear and my questions not too dumb but I'm a bit lost here.

Ben

**Comment by pd3** [comment]
You need to be careful when designing the filtering expressions: you asked to remove (`-e`) variants which have both low quality AND low depth. Perhaps you wanted to remove variants with either low quality OR low depth, so your expression should look like this `-e 'QUAL<20 || DP<10'`. For more tips see this short tutorial http://samtools.github.io/bcftools/howtos/filtering.html

Per-sample allelic depth can be added at the mpileup step by adding the `-a FORMAT/AD` option.

**Comment by KunathBJ** [comment]
Hello,

Is that a problem? I have actually followed the filtering given on the link you provided.

> In this example, we use the -i 'QUAL>20 && DP>10' expression to include only sites with big enough quality and depth

I just excluded them instead. Interestingly enough, even though some variants had a DP lower than 10, they were still assigned as PASS in the final .vcf file.

Thanks for the Per-sample allelic depth, I will look into that right now.

Thanks a lot for your help.
Take care,

Ben
